src/libraries/System.Net.Security/tests/FunctionalTests/NegotiateStreamInvalidOperationTest.cs

```diff
@@ -98,56 +98,6 @@ public async Task NegotiateStream_EndReadEndWriteInvalidParameter_Throws()
             }
         }
 
-        [Fact]
```

Comment:
Why are these tests being deleted?

Reply:
They were testing that if you failed to correctly call EndXx after BeginXx that the instance would be unusable because it would still think an operation was in progress. That's no longer the case, as the instance is always marked as usable again when the async operation completes.

Reply:
>hat's no longer the case, as the instance is always marked as usable again when the async operation completes.

But that is the result of our implementation changes to use async/await in more places as part of making the implementation more efficient internally.

Conceptually, the Begin/End APM pattern says to always call EndXx after BeginXx. Otherwise, the operation is still "in progress".

So, aren't we now changing that pattern?

Reply:
The pattern is still to call it: it's how exceptions are propagated, results are returned, etc. It just no longer invalidates the object if you don't.

Reply:
So, what happens if a user does:

* BeginXx
* BeginXx

i.e. they don't call EndXx.  Do we still get an exception because conceptually we still have an operation in progress?

I'm worried that we are somehow breaking app-compat with this change.  Are we?

Reply:
We generally don't consider taking away failure conditions to be "breaking". Could someone have taken a dependency on it? Sure. But this is the same issue faced anywhere we've flipped begin/end to be on top of async/await with TaskToApm:
https://source.dot.net/#q=Tasktoapm.begin

Reply:
I would consider it as an acceptable breaking change (compared with unacceptable breaking change).

In the .NET Framework world, taking away failure conditions has broken customers in the past because they relied on the exception being generated. I agree that .NET Core has taken a more flexible approach especially with a new major version (.NET 5) being released soon.

I'm fine with "breaking" things. I just want to make sure we document behavior changes like this in some way. We need to remember why we made an intention behavior change. In this case it was because of improving implementation/performance in favor of the async/await implementation.

Reply:
If you want to call it out, we can.  From my perspective it's just inconsistent as we generally haven't called out the removal of error conditions as being breaking.

There's a potentially larger break here, which we should call out (but still take), in that by switching the Begin/EndXx APIs to be on top of the XxAsync APIs rather than the other way around, we end up changing which method calls to which, and these methods are virtuals on a non-sealed type, so in theory someone could be deriving from NegotiateStream and see different behaviors.  In practice I think that's unlikely to be a real issue, given how little NegotiateStream is used, and I've never once seen a derivation from it.  It's also the same issue we've faced any other time we've done this, e.g. with SslStream.

Reply:
I assume this is also true for recent similar changes in SslStream, right?

Reply:
> I assume this is also true for recent similar changes in SslStream, right?

Yes, for the AuthenticateAs methods.  The same applied to SslStream's Read/WriteAsync methods in a previous release when those calls were inverted with their Begin/End counterparts.